integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/context/MgwServerInstance.java

```diff
@@ -118,68 +103,19 @@ public MgwServerInstance(String confPath, boolean tlsEnabled, boolean customJwtT
 
     }
 
-    @Override
-    public void startMGW() throws IOException, InterruptedException {
-        try {
-            environment.start();
-        } catch (Exception e) {
-            log.error("Error occurs when docker-compose up");
-        }
-
-        waitTillBackendIsAvailable();
-    }
-
-    @Override
-    public void stopMGW() {
-        environment.stop();
-    }
-
     /**
      * This will create a separate mgw setup in the target directory to execute the tests.
      *
-     * @throws IOException
+     * @param customJwtTransformerEnabled - whether the custom JWT transformer is enabled or not
      * @throws MicroGWTestException
      */
-    public static void createTmpMgwSetup(boolean customJwtTransformerEnabled) throws IOException, MicroGWTestException {
-        File targetClassesDir = new File(MgwServerInstance.class.getProtectionDomain().getCodeSource().
-                getLocation().getPath());
-        String targetDir = targetClassesDir.getParentFile().toString();
-        final Properties properties = new Properties();
-        properties.load(MgwServerInstance.class.getClassLoader().getResourceAsStream("project.properties"));
-
-        Utils.copyDirectory(targetDir + File.separator + "micro-gwtmp" +  File.separator +
```

Comment:
I think we need this section. Because when the default config gets replaced by a new config.toml and something like a jwt transformer gets added into the dropin folder, the folder that was originally extracted would change. So to get back the original mgw with the default behavior, we will need this section.

Reply:
I have created an abstract parent class and moved the commonly used code there. Please refer to MgwServerImpl.java line #51